src/js/script/subprocess-status.js

```diff
@@ -0,0 +1,88 @@
+(function() {
+  'use strict';
+
+  var JSON_URL = '/subprocess/json/?pk=',
+    id,
+    intervalId,
+    $progressBar,
+    $badge,
+    $completed,
+    $error;
+
+  function getProgress() {
+    $.ajax({
+      url: JSON_URL + id
+    }).done(function (data) {
```

Comment:
The anonymous functions used for the ajax done and failure can be broken out into two separate functions to make the code a bit easy to read.